src/PolyType/Abstractions/ShapeExtensions.cs

```diff
@@ -0,0 +1,32 @@
+﻿using System.ComponentModel;
+
+namespace PolyType.Abstractions;
+
+/// <summary>
+/// Extension methods for the various type shape interfaces.
+/// </summary>
+public static class ShapeExtensions
+{
+    /// <summary>
+    /// Retrieves the default constructor for the given type shape, if available.
+    /// </summary>
+    /// <typeparam name="T">The type to be constructed.</typeparam>
+    /// <param name="shape">The type's object shape.</param>
+    /// <returns>A delegate for the default constructor; or <see langword="null" /> if none is available.</returns>
+    public static Func<T>? GetDefaultConstructor<T>(this IObjectTypeShape<T> shape) => (Func<T>?)shape.Constructor?.Accept(DefaultConstructorVisitor.Instance);
```

Comment:
This can further be extended to collection shapes that have a `Mutable` construction strategy. Perhaps though it just makes sense that we expose this method directly on `ITypeShape<T>` paired with a `HasDefaultConstructor` property?

Reply:
Generic `IEnumerableTypeShape` and `IDictionaryTypeShape` interfaces already define `GetDefaultConstructor()` methods. Implementing this extension method on `ITypeShape` or `ITypeShape<T>` seems to be impossible because those interfaces do not offer a `Constructor` property.

`ITypeShape<T>` being an interface, it cannot be implemented there (while supporting .NET Standard 2.0). And since the method body is the same for all interface implementations, an extension method is the most natural place to put it IMO.
IMO the bool property that predicts whether this method will return `null` doesn't add enough value to add more public API. But I'll go with your suggestion if you insist.